src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/RazorFormattingPass.cs

```diff
@@ -459,6 +460,14 @@ private bool FormatBlock(FormattingContext context, RazorSourceDocument source,
                 changes.Add(new TextChange(source.Text.GetTextSpan(codeRange.End, start), context.NewLineString + additionalIndentation));
                 didFormat = true;
             }
+
+            // If there is no newline after the close brace, we add one.
```

Comment:
I might be smooth braining but this doesn't check if a newline exists it just always adds one. I assume that's fine but scratching my head a little

Reply:
It's a bit hidden, I'll add a comment. It's the call to `GetFirstNonWhitespaceOffset`, starting from the end of the close brace. If there is only whitespace after the brace then it will return null, and we won't do anything.

Reply:
Ohhh. Because if there was a non whitespace character after then obviously no newline